tensorizer/_NumpyTensor.py

```diff
@@ -0,0 +1,297 @@
+        """
+        try:
+            return _INTERMEDIATE_MAPPING[size]
+        except KeyError as e:
+            raise ValueError(
+                "Cannot create a numpy array with opaque elements of size"
+                f" {size} bytes"
+            ) from e
+
+    @classmethod
+    def _is_opaque(cls, numpy_dtype: str) -> bool:
+        """
+        A check to see if the dtype needs to be swapped while decoding,
+        based on whether the encoded dtype is in the opaque format
+        used by this class.
+
+        Args:
+            numpy_dtype: The numpy dtype, as encoded in a tensorized file.
+
+        Returns:
+            True if the encoded dtype is opaque, False otherwise.
+        """
+        return numpy.dtype(numpy_dtype).type == numpy.void
+
+    @classmethod
+    def _is_asymmetric(cls, torch_dtype: torch.dtype) -> bool:
+        """
+        A check to see if the dtype needs to be swapped while encoding,
+        based on whether numpy has a corresponding dtype or not.
+        This check is hardcoded, not dynamic, but up to date as of torch 2.0.
+
+        Args:
+            dtype: The torch dtype to check
+
+        Returns:
+            True if a class is known not to have a corresponding numpy dtype,
+            False otherwise.
+        """
+        return torch_dtype in _ASYMMETRIC_TYPES
+
+    @classmethod
+    def _decoder_dtype(cls, numpy_dtype: str):
+        """
+        Converts an opaque storage numpy dtype generated by this class
+        into one that numpy can properly decode.
+
+        NB: Even though a dtype like ``numpy.dtype("<V2")`` is valid,
+        referring to the ``void16`` pseudo-type, numpy does not respect
+        the endianness indicated in the type string when loading this way.
+        If changed from ``<V2`` to ``<i2`` to load it as an int, it works fine.
+
+        Args:
+            numpy_dtype: The encoded numpy dtype.
+
+        Returns:
+            A dtype suitable for passing to numpy
+        """
+        if cls._is_opaque(numpy_dtype):
+            return numpy_dtype.replace("V", "i")
+        else:
+            return numpy_dtype
+
+    def _decode_torch_dtype(self) -> torch.dtype:
+        """
+        Parses the `self.torch_dtype` field.
+
+        Returns: An instance of ``torch.dtype`` corresponding to the string
+            stored in `self.torch_dtype`.
+
+        Raises:
+            ValueError: If `self.torch_dtype` is not set, is not in the form
+                "torch.<dtype>", cannot be found in torch, or refers to
+                something other than a ``torch.dtype``.
+            TypeError: If `self.torch_dtype` is not a string.
+        """
+        # Quick route, table lookup for common types
+        dtype = _DECODE_MAPPING.get(self.torch_dtype)
+        if dtype is not None:
+            return dtype
+        else:
+            # Long route using getattr(), any other type
+            if not self.torch_dtype:
```

Comment:
@Eta0 
nit: I'm not a fan of overly indented python code, especially with smaller line lengths.  In cases like this where there's a return in the `if`, you don't technically need an `else` and can save the indentation.

Thoughts? Do you think it's more readable keeping the `else`?

Reply:
I have no preference here. Since the fact that they are two separate routes is already commented I think it would be perfectly readable unwrapping the `else`.